AndroidAnnotations/androidannotations-core/androidannotations-test/src/test/java/org/androidannotations/test/menu/InjectMenuActivityTest.java

```diff
@@ -51,14 +58,34 @@ public void testMenuInjectedFromOnCreateOptionsMenu() {
 	@Test
 	public void methodInjectedMenu() {
 		Menu menu = mock(Menu.class);
+
+		assertThat(injectMenuActivity.menuIsInflated).isFalse();
 		injectMenuActivity.onCreateOptionsMenu(menu);
+		assertThat(injectMenuActivity.menuIsInflated).isTrue();
+
 		assertThat(injectMenuActivity.methodInjectedMenu).isSameAs(menu);
 	}
 
 	@Test
 	public void multiInjectedMenu() {
 		Menu menu = mock(Menu.class);
+
+		assertThat(injectMenuActivity.menuIsInflated).isFalse();
 		injectMenuActivity.onCreateOptionsMenu(menu);
+		assertThat(injectMenuActivity.menuIsInflated).isTrue();
+
 		assertThat(injectMenuActivity.multiInjectedMenu).isSameAs(menu);
 	}
+
+	private InjectMenuActivity.MockMenuInflater createMenuInflater() {
+		final InjectMenuActivity.MockMenuInflater menuInflater = Mockito.mock(InjectMenuActivity.MockMenuInflater.class);
+		doAnswer(new Answer<Void>() {
+			public Void answer(InvocationOnMock invocation) {
+				menuInflater.menuInflated = true;
```

Comment:
Why is this needed? You can just you Mockito verification, since you already using mocking.
```java
Mockito.verify(menuInflater).inflate(anyInt(), any(Menu.class));
```

This way you can get rid of the inner class and the boolean field as well.

Another thing: most of the times, we static import methods like `mock` and `verify`.